### PR TITLE
Framework: add site.vouchers to redux

### PR DESCRIPTION
**Pull request by retrofox**
This PR adds `site.vouchers` to the global state tree. For now It isn't being used in any part of the app.

### Testing instructions

* No functional changes
* tests pass `npm run test-client client/state/sites/vouchers/test/*.js`

image

### CODE REVIEW

**Comment by retrofox** [comment]
cc @gwwar

File: client/state/sites/vouchers/actions.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	SITE_VOUCHERS_RECEIVE,
+	SITE_VOUCHERS_REQUEST,
+	SITE_VOUCHERS_REQUEST_SUCCESS,
+	SITE_VOUCHERS_REQUEST_FAILURE,
+} from 'state/action-types';
+
+/**
+ * Module vars
+ */
+const errorMessage = 'There was a problem fetching site vouchers. Please try again later or contact support.';
```
Comment (gwwar) [inline client/state/sites/vouchers/actions.js]: Will this be seen by users? Maybe return an error code, and let the component translate the message.

File: client/state/sites/vouchers/selectors.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @param {Object} state - current state object
+ * @param {Number} siteId - site identificator
+ * @return {Array} site vouchers
+ */
+export const getVouchersBySiteId = ( state, siteId ) => {
+	if ( ! siteId ) {
+		return [];
+	}
+
+	const { items } = state.sites.vouchers;
```
Comment (gwwar) [inline client/state/sites/vouchers/selectors.js]: How about:
```javascript
return state.sites.vouchers.items[ siteId ] || [];
```

File: client/state/sites/vouchers/actions.js (hunk 1 above)
Comment (retrofox) [inline client/state/sites/vouchers/actions.js]: I think we should directly remove the message.

File: client/state/sites/vouchers/selectors.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Return requesting state for the given site
+ *
+ * @param {Object} state - current state object
+ * @param {Number} siteId - site identifier
+ * @return {Boolean} is site-vouchers requesting?
+ */
+export const isRequestingSiteVouchers = ( state, siteId ) => {
+	const { requesting } = state.sites.vouchers;
```
Comment (gwwar) [inline client/state/sites/vouchers/selectors.js]: Maybe:
```javascript
return state.sites.vouchers.requesting[ siteId ] || false;
```

File: client/state/sites/vouchers/selectors.js (hunk 2 above)
Comment (retrofox) [inline client/state/sites/vouchers/selectors.js]: 👍

**Comment by gwwar** [comment]
We'll also want to add vouchers to `state/sites/reducer.js` and update the test there for expected reducer keys. Once you've done that, should be good to 🚢

**Comment by retrofox** [comment]
> We'll also want to add vouchers to state/sites/reducer.js and update the test there for expected reducer keys. Once you've done that, should be good to 🚢

Right, I already implemented in another PR but you're right.. It must be added in this one.

**Comment by retrofox** [comment]
`npm run test-client client/state/sites/test/* -- -w`

image

**Pull request merged**